docs/root/configuration/http/http_conn_man/headers.rst

```diff
@@ -9,6 +9,22 @@ is being received) as well as during encoding (when the response is being sent).
 .. contents::
   :local:
 
+.. _config_http_conn_man_headers_scheme:
+
+:scheme
+-------
+
+Envoy will always set the *:scheme* header while processing a request. It should always be available to filters, and should be forwarded upstream for HTTP/1 and HTTP/2, where :ref:`config_http_conn_man_headers_x-forwarded-proto` will be sent for HTTP/1.1
+
+For HTTP/2, and HTTP/3, incoming :scheme headers are trusted and propogated through upstream.
+For HTTP/1, the :scheme header will be set
+1) From the absolute URL if present and valid. An invalid (not "http" or "https") scheme, or an https scheme over an unencrypted connection will result in Envoy rejecting the request.
```

Comment:
Rejection for not matching scheme happens for H2 and H3 also, right? Clarify those cases?

Reply:
Is this a new behaviors?  or envoy has always been enforcing ":scheme" header?

Reply:
> not sure how much of this detail it makes sense to have in these docs, thoughts from you two?

I myself didn't realized we don't check this for H2. As an aside it seems like we should, since there is nothing to prevent people from sending https scheme over a plaintext H2 connection, right? Either way I would personally overshare here since it's so complicated and important.

Reply:
If it is enforcing `scheme` from absolute URL,  I am fine.  If it is enforcing the `:scheme` header with downstream connection security level.  I do have concerns. Many proxy systems choose to preserve downstream `:scheme`, it may result in inconsistency between `:scheme` header and connection security level.